pkg/scheduler/api/types.go

```diff
@@ -28,9 +28,13 @@ import (
 	"github.com/NVIDIA/KAI-scheduler/pkg/scheduler/api/resource_info"
 )
 
+type Partition []*node_info.NodeInfo
+
 // PredicateFn is used to predicate node for task.
 type PredicateFn func(*pod_info.PodInfo, *podgroup_info.PodGroupInfo, *node_info.NodeInfo) error
 
+type PartitionFn func(*podgroup_info.PodGroupInfo, Partition) ([]Partition, error)
```

Comment:
I understand why the first parameter is the job since it dictates how nodes are partitioned, why the second parameter is a Partition? Maybe it should simply be a slice of all cluster nodes?